tree_in_context/src/interpreter.rs

```diff
@@ -263,11 +310,14 @@ impl<'a> VirtualMachine<'a> {
                 vals[*i].clone()
             }
             // in_context this is type checked, so ignore type
-            Expr::Alloc(e_size, _ignore_state, _ty) => {
+            Expr::Alloc(e_size, state_expr, _ty) => {
                 let size = self.interp_int_expr(e_size, arg);
+                let state_val = self.interpret_expr(state_expr, arg);
+                assert_eq!(state_val, Value::StateV);
                 let addr = self.next_addr;
                 self.next_addr += usize::try_from(size).unwrap();
-                Ptr(Pointer::new(addr, size as usize, 0))
+
+                tuplev!(Ptr(Pointer::new(addr, size as usize, 0)), Value::StateV)
```

Comment:
what's the 0 for?

Reply:
0 is the offset for the pointer (alloc returns a new memory region with a pointer pointing to the start of it)
Added a comment